src/components/encounter/encounter-form-manager.ts

```diff
@@ -106,6 +122,38 @@ export class EncounterFormManager {
       return savePatientIdentifier(patientIdentifier, patient.id);
     });
   }
+
+  static saveProgramEnrollments = (programPayload: ProgramEnrollmentPayload, sessionMode: string, patient: Patient) => {
+    const ac = new AbortController();
+    if (programPayload.program) {
+      let patientProgramEnrollment = patient.patientPrograms.find(
+        (enrollment) => enrollment.program.uuid === programPayload.program && enrollment.dateCompleted === null,
+      );
+
+      // return already enrolled if not edit mode
+      if (sessionMode == 'edit' && patientProgramEnrollment?.uuid) {
```

Comment:
if there is no uuid on the payload, it means its a new enrollment

Reply:
It would make the code more clear if we explicitly defined variables for `currentEnrolment` (picked from the DB) vs `newEnrolment` (established within the form session)